src/animals/__test__/animals.controller.spec.ts

```diff
@@ -1,38 +1,42 @@
-import { Test, TestingModule } from '@nestjs/testing';
-import { ParamsWithId } from 'src/validations/id.validator';
+import { Test } from '@nestjs/testing';
+import { AnimalType } from 'src/types/animals.type';
 import { AnimalsController } from '../animals.controller';
 import { AnimalsService } from '../animals.service';
 import { Animal } from '../schemas/animal.schema';
 
-jest.mock('../animals.service');
+const animalStub: Animal = {
+  animalName: 'Human',
+  type: AnimalType.MAMMALS,
+  createdAt: new Date('2023-03-19T18:27:12.933Z'),
+  description: 'test description',
+};
 
 describe('AnimalsController', () => {
-  let controller: AnimalsController;
-  let service: AnimalsService;
+  let animalsController: AnimalsController;
+  let animalsService: AnimalsService;
 
   beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
+    const moduleRef = await Test.createTestingModule({
       controllers: [AnimalsController],
-      providers: [AnimalsService],
+      providers: [
+        AnimalsService,
+        {
+          provide: AnimalsService,
+          useValue: {
+            findAll: jest.fn().mockResolvedValue([animalStub]),
+          },
+        },
+      ],
     }).compile();
 
-    controller = module.get<AnimalsController>(AnimalsController);
-    service = module.get<AnimalsService>(AnimalsService);
-    jest.clearAllMocks();
+    animalsController = moduleRef.get<AnimalsController>(AnimalsController);
+    animalsService = moduleRef.get<AnimalsService>(AnimalsService);
   });
 
-  describe('getAnimal', () => {
-    describe('when getAnimal is called', () => {
-      let animals: Animal[];
-
-      beforeEach(async () => {
-        const id = new ParamsWithId();
-        animals = await controller.getAllAnimals();
-      });
-
-      it('should call animalsService', () => {
-        expect(service.findAll).toBeCalled();
-      });
+  describe('getAllAnimals', () => {
```

Comment:
brakuje testu na zle dane

Reply:
Ten endpoint w sumie tylko ściąga wszystkie elementy z BD. Więc jedynie co może wyskoczyć to błąd Mongodb. Wtedy Nest to wyłapie i rzuci Internal Server Error. Czy to mam sprawdzić?